FreeRTOS-Plus/Source/Application-Protocols/network_transport/using_mbedtls_pkcs11/using_mbedtls_pkcs11.c

```diff
@@ -634,8 +639,8 @@ static CK_RV initializeClientKeys( SSLContext_t * pxCtx )
     {
         /* Get the handle of the device private key. */
         xResult = xFindObjectWithLabelAndClass( pxCtx->xP11Session,
-                                                pkcs11configLABEL_DEVICE_PRIVATE_KEY_FOR_TLS,
-                                                sizeof( pkcs11configLABEL_DEVICE_PRIVATE_KEY_FOR_TLS ) - 1UL,
+                                                pcLabelName,
+                                                strlen( pcLabelName ),
```

Comment:
```suggestion
                                                strnlen( pcLabelName, pkcs11configMAX_LABEL_LENGTH ),
```

Reply:
There's a similar problem on line 428. Unless the string is const char * const or clearly a #defined constant, you should avoid strlen.

Reply:
https://github.com/FreeRTOS/FreeRTOS/pull/723/commits/e5d9e51d11d2047e54f0cd1dbe283b8de0f320a5

Swapped strlen for strnlen in both places in this commit.